code/datums/components/persistent_overlay.dm

```diff
@@ -0,0 +1,33 @@
+/datum/component/persistent_overlay
+	var/image/persistent_overlay = null
+	var/atom/target = null
+
+/datum/component/persistent_overlay/Initialize(image_overlay, _target, timer)
+	persistent_overlay = image_overlay
+	target = _target
```

Comment:
If `target` gets deleted somehow without the `parent` being deleted, then the reference will stay here. Be sure to also register to the `target` being deleted to remove the component. Or handle it in another way